apps/src/studio/studio.js

```diff
@@ -371,6 +375,10 @@ function collisionTest(x1, x2, xVariance, y1, y2, yVariance) {
   return (Math.abs(x1 - x2) <= xVariance) && (Math.abs(y1 - y2) <= yVariance);
 }
 
+function overlappingTest(x1, x2, xVariance, y1, y2, yVariance) {
```

Comment:
just for trivia, here's my favourite collision function again.  (tests for two rectanges overlapping or not by actually detecting a miss)

function Collide(x1, y1, w1, h1, x2, y2, w2, h2)
{
  // detect a non-collision.  Otherwise we have a collision.

  if (x1 + w1 - 1 < x2 ||
      x1 > x2 + w2 - 1 ||
      y1 + h1 - 1 < y2 ||
      y1 > y2 + h2 - 1)
    return false;

  return true;
}